cincinnati/src/plugins/catalog.rs

```diff
@@ -0,0 +1,65 @@
+//! Plugins catalog.
+//!
+//! This catalog relies on a static list of all available plugins,
+//! referenced by name. It is used for configuration purposes.
+
+use super::internal::channel_filter::ChannelFilterPlugin;
+use super::internal::edge_add_remove::EdgeAddRemovePlugin;
+use super::internal::metadata_fetch_quay::QuayMetadataFetchPlugin;
+use super::internal::node_remove::NodeRemovePlugin;
+use super::{Plugin, PluginIO};
+use failure::Fallible;
+use std::fmt::Debug;
+
+/// Key used to look up plugin-type in a configuration entry.
+static CONFIG_PLUGIN_NAME_KEY: &str = "name";
+
+/// Settings for a plugin.
+pub trait PluginSettings: Debug + Send {
+    /// Build the corresponding plugin for this configuration.
+    fn build_plugin(&self) -> Fallible<Box<Plugin<PluginIO>>>;
```

Comment:
That would be nice, however I'm not sure that's possible. In particular, taking the ownership and using it results in:
```
error[E0161]: cannot move a value of type dyn plugins::catalog::PluginSettings:
 the size of dyn plugins::catalog::PluginSettings cannot be statically determined
```
And adding a `Sized` bound turns into:
```
error[E0038]: the trait `plugins::catalog::PluginSettings` cannot be made into an object
   = note: the trait cannot require that `Self : Sized`
```

Reply:
It does seem to me like we're not making full use of the type system here since two of the methods which need to be implemented by every plugin aren't in any trait, so it's mere convention. However I'm failing to come up with something that seems better :man_shrugging: